### PR TITLE
feat: added push notification screen in onboarding and settings

### PR DESCRIPTION
**Pull request by wadeking98**
# Summary of Changes

Added Screen to enable/disable push notifications. Added logic to handle when notifications are disabled at the OS level

## Push Notifications during onboarding:
https://github.com/openwallet-foundation/bifold-wallet/assets/36937407/e10b2c4b-af08-4964-910c-57e28a8fe9ac

## Push Notifications prompt on existing app (not fresh install):
https://github.com/openwallet-foundation/bifold-wallet/assets/36937407/35509b62-7ba1-4a6a-9d34-d30c089cfdb6

## Push Notifications from settings:
https://github.com/openwallet-foundation/bifold-wallet/assets/36937407/ca0c6abd-71bf-4908-9f91-378847fc6c63

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/App/contexts/configuration.tsx [hunk 1]
```diff
@@ -51,6 +61,7 @@ export interface ConfigurationContext {
   showPreface?: boolean
   disableOnboardingSkip?: boolean
   useBiometry: React.FC
+  pushNotification?: PushNotificationConfiguration
```
Comment (jleach) [inline packages/legacy/core/App/contexts/configuration.tsx]: @cvarjao Ok or should we be using dependency injection now?

File: packages/legacy/core/App/localization/en/index.ts [hunk 2]
```diff
@@ -297,6 +297,22 @@ const translation = {
...
+  "PushNotifications": {
+    "EnableNotifiactions": "Enable Notifications to get instant alerts",
+    "BeNotified": "Be notified when you receive:",
+    "ReceiveNotifications": "Receive notifications",
+    "BulletFour": "new messages",
+    "BulletOne": "new credential offers",
+    "BulletThree": "updates to your credentials",
+    "BulletTwo": "new proof requests",
+    "NotificationsOffMessage": "Be notified when you receive new credential offers, proof requests, updates to your credentials, new messages and more.",
+    "NotificationsOffTitle": "Notifications for BC Wallet are turned off",
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/localization/en/index.ts]: I think you meant to put this in BC Wallet, or meant to make it generic to Bifold and then override in BC Wallet.

**Review by jleach — DISMISSED** [review]

File: packages/legacy/core/App/navigators/RootStack.tsx [hunk 3]
```diff
@@ -46,8 +46,14 @@ const RootStack: React.FC = () => {
...
   const defaultStackOptions = createDefaultStackOptions(theme)
-  const { splash, showPreface, enableImplicitInvitations, enableReuseConnections, enableUseMultUseInvitation } =
-    useConfiguration()
+  const {
+    splash,
+    showPreface,
+    enableImplicitInvitations,
+    enableReuseConnections,
+    enableUseMultUseInvitation,
+    pushNotification,
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/navigators/RootStack.tsx]: Should we rename this `enablePushNotifications` or something like that? What do you think

**Review by bryce-mcmath — CHANGES_REQUESTED** [review]

File: packages/legacy/core/App/navigators/RootStack.tsx (hunk 3 above)
Comment (wadeking98) [reply packages/legacy/core/App/navigators/RootStack.tsx]: ya I think that would be a good idea

**Review by bryce-mcmath — APPROVED** [review]

**Pull request merged**